src/app/core/http-services/item-navigation.service.ts

```diff
@@ -199,4 +199,45 @@ export class ItemNavigationService {
     return (isSkill(type)) ? this.getRootSkills() : this.getRootActivities();
   }
 
+  getNavigationNeighbors(itemRoute: ItemRoute): Observable<{ parent: ItemRoute|null, left: ItemRoute|null, right: ItemRoute|null }> {
+    const parentId = itemRoute.path[itemRoute.path.length - 1];
+
+    // Root activity => no parent/left/right activity
+    if (!parentId) return of({ parent: null, left: null, right: null });
+
+    return this.getNavDataFromChildRoute(parentId, itemRoute).pipe(map(navParent => {
+      const index = navParent.items.findIndex(item => item.id === itemRoute.id);
+      if (index === -1) throw new Error(`Unexpected: item is missing from its parent children list`);
+
+      const parentAttemptId = navParent.parent.attemptId;
+      if (parentAttemptId === null) throw new Error('Unexpected: parent of an item node has no attempt');
+
+      const typeCat : ItemTypeCategory = itemRoute.contentType === 'activity' ? 'activity' : 'skill';
```

Comment:
I don't get it, you convert a `ItemTypeCategory` to a `ItemTypeCategory`? 🤔

Reply:
fixed